libraries/AP_OSD/AP_OSD.h

```diff
@@ -190,10 +233,90 @@ class AP_OSD_Screen
     void draw_clk(uint8_t x, uint8_t y);
 };
 
+/*
+  class to hold one setting
+ */
+class AP_OSD_ParamSetting : public AP_OSD_Setting
+{
+public:
+    // configured index.
+    AP_Int16 index;
+    // parameter number
+    uint8_t _number;
+    char _name[17];
+    AP_Param* param;
+    ap_var_type type;
+
+    AP_OSD_ParamSetting(uint8_t param_number, bool enabled, uint8_t x, uint8_t y, uint16_t idx);
+    AP_OSD_ParamSetting(uint8_t param_number, bool enabled, uint8_t x, uint8_t y, const char* name);
+
+    // initialize the setting from the configured information
+    void update();
+
+    // User settable parameters
+    static const struct AP_Param::GroupInfo var_info[];
+
+private:
+    uint16_t _current_index;
+};
+
+/*
+  class to hold one screen of settings
+ */
+class AP_OSD_ParamScreen : public AP_OSD_AbstractScreen
+{
+public:
+    enum class Event {
+        NONE,
+        MENU_ENTER,
+        MENU_UP,
+        MENU_DOWN,
+        MENU_EXIT
+    };
+
+    static const uint8_t NUM_PARAMS = 8;
+    // constructor
+    AP_OSD_ParamScreen();
+
+    void draw(void) override;
+
+    // User settable parameters
+    static const struct AP_Param::GroupInfo var_info[];
+
+private:
+    AP_OSD_ParamSetting params[NUM_PARAMS] = {
```

Comment:
Shouldn't this be `static const` somehow?

Reply:
These are just the initial defaults. I'm not sure how I would make these static without using up more memory